Dockerfile.multistage

```diff
@@ -0,0 +1,24 @@
+# Start from golang v1.10.4 base image
+FROM golang:1.10.4 as builder
+# Workdir is path in your docker image from where all your commands will be executed
+WORKDIR /go/src/app
+# Copy all from your project inside workdir of docker image
+COPY . .
+# Download dependencies
+RUN go get -d -v ./...
+# Install the package
+RUN go install -v ./...
+# Build the Go app
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /go/bin/app .
+
+######## Start a new stage from scratch #######
+FROM alpine:latest  
```

Comment:
you can use scratch. And don't use latest, always bind to some specific version. Only in that way you will have reproducible builds